app/tests/content/test_badge_integration.py

```diff
@@ -0,0 +1,160 @@
+
+from app.content.factories.badge_factory import BadgeFactory
+from app.util.test_utils import get_api_client
+from app.util.utils import now
+
+API_BADGE_BASE_URL = "/badges/"
+
+
+def _get_badge_leaderboard_url():
+    return f"{API_BADGE_BASE_URL}leaderboard/"
+
+
+def _get_badge_specific_leaderboard_url(badge):
+    return f"{API_BADGE_BASE_URL}{badge.id}/leaderboard/"
+
+
+def _get_user_badge_url():
+    return f"/users/me{API_BADGE_BASE_URL}"
+
+
+def _get_badge_flag(badge):
+    return {"flag": badge.flag}
+
+
+@pytest.mark.django_db
+def test_get_badge_leaderboard_as_anonymous_user(default_client):
+    """An anonymous user should not be able to see the leaderboard"""
+
+    url = _get_badge_leaderboard_url()
+    response = default_client.get(url)
+
+    assert response.status_code == status.HTTP_403_FORBIDDEN
+
+
+@pytest.mark.django_db
+def test_get_badge_spesific_leaderboard_as_anonymous_user(default_client, badge):
+    """An anonymous user should not be able to see badge-specific leaderboard"""
+
+    url = _get_badge_specific_leaderboard_url(badge)
+    response = default_client.get(url)
+
+    assert response.status_code == status.HTTP_403_FORBIDDEN
+
+
+@pytest.mark.django_db
+def test_get_request_for_user_badges_as_anonymous_user(default_client):
+    """An anonymous user should not be able to do a get request for a user badge"""
+
+    url = _get_user_badge_url()
+    response = default_client.get(url)
+
+    assert response.status_code == status.HTTP_403_FORBIDDEN
+
+
+@pytest.mark.django_db
+def test_create_user_badge_as_member(member, badge):
+    """All members should be able to create a user badge for themselves"""
+
+    url = _get_user_badge_url()
+    client = get_api_client(user=member)
+    data = _get_badge_flag(badge)
+    response = client.post(url, data)
+
+    assert response.status_code == status.HTTP_200_OK
+
+
+@pytest.mark.django_db
+def test_create_user_badge_for_already_existing_user_badges(member, badge):
+    """User badges that already exist can not be created and throws 409 unique-together exception"""
+
+    url = _get_user_badge_url()
+    client = get_api_client(user=member)
+    data = _get_badge_flag(badge)
+    client.post(url, data)
+    response = client.post(url, data)
+    detail = response.json().get("detail")
+
+    assert response.status_code == status.HTTP_409_CONFLICT
+    assert detail == "Some values are supposed to be unique but are not."
+
+
+@pytest.mark.django_db
+def test_create_user_badge_for_non_existent_badge(member):
+    """Badge has to exist for a User Badge to be created. Therefore the view returns a 404"""
+
+    url = _get_user_badge_url()
+    client = get_api_client(user=member)
+    data = {"flag": "flag_unlike_anything_else"}
+    response = client.post(url, data)
+
+    assert response.status_code == status.HTTP_404_NOT_FOUND
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize(
+    ("active_from", "active_to", "expected_status_code", "expected_detail"),
```

Comment:
Du trenger ikke sjekke `expected_detail` når `expected_status_code` i praksis sjekker det samme. I tillegg er det ikke usannsynlig at `detail` vil endres en gang, men de breaker ingenting og da trenger vi heller ikke teste det
```suggestion
    ("active_from", "active_to", "expected_status_code"),
```